README.md

```diff
@@ -4,3 +4,119 @@
 # MQTT bridge for Apache Kafka®
 
 This project provides a software component which acts as a bridge between [MQTT 3.1.1](http://docs.oasis-open.org/mqtt/mqtt/v3.1.1/os/mqtt-v3.1.1-os.html) and an [Apache Kafka®](https://kafka.apache.org/) cluster.
+
+## Running the bridge
+
+### Locally
+
+Clone the repository and update your dependencies using maven.
+Afterward, edit the `config/application.properties` file which contains the configuration. And provide a valid [topic mapping rules file]().
+
+Once your configuration is ready, start the bridge using:
+
+```bash
+<application> --config-file config/application.properties --mapping-rules=path/to/mapping-rules
+```
+
+You can find more information about the topic mapping rules in the MQTT Bridge.
+
+### Running Tests
+```bash
+mvn clean test
+```
+
+## MQTT bridge documentation
+
+### 1. MQTT Bridge Overview
+You can use the MQTT bridge to take advantage of the benefits of both MQTT and Kafka in data processing and IoT system scenarios.
+
+#### 1.1 General overview.
+
+#### 1.2 How it works
+
+The MQTT Bridge is a netty-based MQTT server that is responsible for handling all the MQTT client connections, connections management, and all MQTT-based operations like QoS, error handling, and other capabilities which Netty offers.
+
+The MQTT Bridge provides two interfaces to integrate MQTT and Apache Kafka. 
+1. MQTT mapper: maps incoming MQTT topics to Kafka topics. 
+2. MQTT Bridge:  transforms the incoming MQTT payload to a valid Kafka record and then sends them to the appropriate Kafka topic using a Kafka Producer.
```

Comment:
for me the MQTT bridge is the entire application, it's not what you are explaining here. It's the entire bridge, including the MQTT server, the topic mapper, the Kafka producer and so on.